pkg/cluster/deploybaseresources.go

```diff
@@ -178,6 +185,54 @@ func (m *manager) deployBaseResourceTemplate(ctx context.Context) error {
 	return arm.DeployTemplate(ctx, m.log, m.deployments, resourceGroup, "storage", t, nil)
 }
 
+// subnetsWithServiceEndpoint returns a unique slice of subnet resource IDs that have the corresponding
+// service endpoint
+func (m *manager) subnetsWithServiceEndpoint(ctx context.Context, serviceEndpoint string) ([]string, error) {
+	subnetsMap := map[string]struct{}{}
```

Comment:
why a map here

Reply:
returns a set of subnets.  

if you try to create NACLs with two of the same subnet it will fail.

Reply:
yeah, the alternative is to loop over the slice looking for a subnet already added..